docs/gherkin.md

```diff
@@ -0,0 +1,211 @@
+# Gherkin
+
+Gherkin is the language that Cucumber understands. It is a Business Readable, [Domain Specific Language](http://martinfowler.com/bliki/BusinessReadableDSL.html) that lets you describe software’s behaviour without detailing how that behaviour is implemented.
+
+Gherkin serves two purposes — documentation and automated tests. The third is a bonus feature — when it yells in red it’s talking to you, telling you what code you should write.
+
+Gherkin’s grammar exists in different flavours for many spoken languages (37 at the time of writing), so that your team can use the keywords in your own language.
+
+There are a few conventions.
+
+ * Single Gherkin source file contains a description of a single feature.
+ * Source files have `.feature` extension.
+
+## Gherkin Syntax
+
+A Gherkin source file usually looks like this
+```
+Feature: Some terse yet descriptive text of what is desired
+  Textual description of the business value of this feature
+  Business rules that govern the scope of the feature
+  Any additional information that will make the feature easier to understand
+
+  Scenario: Some determinable business situation
+    Given some precondition
+    And some other precondition
+    When some action by the actor
+    And some other action
+    And yet another action
+    Then some testable outcome is achieved
+    And something else we can check happens too
+
+  Scenario: A different situation
+    ...
+```
+
+First line starts the feature. Lines 2–4 are unparsed text, which is expected to describe the business value of this feature. Line 6 starts a scenario. Lines 7–13 are the steps for the scenario. Line 15 starts next scenario and so on.
```

Comment:
I'd avoid referring to line numbers unless we can display them in the margin somehow. Maybe GitBook has a plugin for that?